scripts/download/tools.ts

```diff
@@ -3,15 +3,49 @@ import fs from 'fs';
 import os from 'os';
 import path from 'path';
 
-import { download, downloadZip, downloadTarGZ, getResource } from '../lib/download.mjs';
+import {
+  download, downloadZip, downloadTarGZ, getResource, DownloadOptions,
+} from '../lib/download';
+import DependencyVersions from './dependencies';
+
+type DependencyPlatform = 'wsl' | 'linux' | 'darwin' | 'win32';
+type Platform = 'linux' | 'darwin' | 'win32';
+type KubePlatform = 'linux' | 'darwin' | 'windows';
+
+type DownloadContext = {
+  dependencyPlaform: DependencyPlatform;
+  platform: Platform;
+  kubePlatform: KubePlatform;
+  // Difference between k8s world and docker compose makes this difficult.
+  // So instead, we determine arch inside the download function.
+  // arch: 'amd64' | 'arm64';
```

Comment:
Hmm, we could at least extract out the `process.env.M1` check to `isArm64: boolean`, maybe?

Reply:
I like the idea of the `isArm64` property, but I think `isArm64` is misleading - we may have other operating systems using `arm64 in the future`. I think I will call it `isM1` for now - let me know if you would prefer a different name.

Reply:
We can revisit once we write support for windows/arm64 or linux/arm64 :)